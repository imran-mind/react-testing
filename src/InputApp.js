import React from 'react';
import { Input } from './Input';

function InputApp() {
    const [inputValue, setInputValue] = React.useState('');
  
    const handleChange = (event) => {
      setInputValue(event.target.value);
    };
  
    return (
      <div>
        <h1>Add Name :</h1>
        <Input handleChange={handleChange} inputValue={inputValue} />
      </div>
    );
}

export default InputApp;