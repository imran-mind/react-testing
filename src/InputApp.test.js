import React from 'react';
import {render,screen} from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import InputApp from './InputApp';
import { Input } from './Input';

describe('Testing InputApp component',()=>{

  test('Input value is updared correctly ',()=>{
    render(<InputApp/>);
    // screen.debug()
    const input = screen.getByRole('textbox');
    UserEvent.type(input,'React');
    expect(input.value).toBe('React');
  });

  test('Input value click count ',()=>{
    const handleChange = jest.fn();

    render(<Input handleChange={handleChange} inputValue=""/>);
    const input = screen.getByRole('textbox');
    UserEvent.type(input,'React');

    expect(handleChange).toHaveBeenCalledTimes(5);
  })

  
  test('snapshot test',()=>{
    const tree = renderer.create(<InputApp/>).toJSON();
    expect(tree).toMatchSnapshot()

  })

})