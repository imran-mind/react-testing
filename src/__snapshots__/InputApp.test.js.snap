// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputApp component snapshot test 1`] = `
<div>
  <h1>
    Add Name :
  </h1>
  <input
    onChange={[Function]}
    value=""
  />
</div>
`;
